include/rear-panel/core/messages.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+#include <cstdint>
+#include <variant>
+
+namespace messages {
+
+struct IncomingMessageFromHost {
+    const char* buffer;
+    const char* limit;
+};
+
+// TODO(ryan): as part of RET-1307 create an auto gen of the ids like the can
+// messages and include them from another file
+enum class MessageType : uint16_t {
+    ECHO = 0x00,
+};
+
+template <MessageType MType>
+struct BinaryFormatMessage {
+    static const auto message_type = MType;
+    const uint16_t length;
+    const char* data;
```

Comment:
is the idea that this data will physically remain in the double buffer until it's done processing, which maybe is just until the message is received? if we want to copy it out of the double buffer immediately we'll need this to be a `std::array<char>` of appropriate max size.

Reply:
Also if these are binary messages it's a bad idea IMO to put them in a native stringlike type because you reasonably read a native stringlike type as zero-terminated which these definitely won't be. at the very least imo this should be `uint8_t*`.